File: hardware/c_code/02_watering_system/02_watering_system.c
#include <wiringPi.h>

#define SENSOR_PIN 0 // 17
#define RELAY_PIN  1//18
#define REED_PIN  3 //22
#define LED_PIN 2//27
#define IRO_PIN 21 //5
#define LED_PIN_IR 5 // 24

int main(void)
{
  if (wiringPiSetup() == -1) {
    return 1;
  }

  pinMode(SENSOR_PIN, INPUT);
  pinMode(RELAY_PIN, OUTPUT);
  pinMode(REED_PIN, INPUT);
  pinMode(LED_PIN, OUTPUT);
  pinMode(LED_PIN_IR,OUTPUT);

  while (1) {
    int sensor_value = digitalRead(SENSOR_PIN);
    int reed_value = digitalRead(REED_PIN);
    int ir_val = digitalRead(IRO_PIN);

    if (sensor_value == HIGH) {
        digitalWrite(RELAY_PIN, LOW);
    } else {
        digitalWrite(RELAY_PIN, HIGH);
    }

    if (reed_value == HIGH) {
        digitalWrite(LED_PIN, HIGH);
    } else {
        digitalWrite(LED_PIN, LOW);
    }

    if ( ir_val == HIGH){
        digitalWrite(LED_PIN_IR,HIGH);
    } else {
        digitalWrite(LED_PIN_IR,LOW);
    }
    
  }

  return 0;
}

File: hardware/python_code/python_code/02_watering_system.py
import RPi.GPIO as GPIO
import time

from pymongo import MongoClient

def update_document(my_device, status):
    uri = "mongodb+srv://Home_Automation_AUT:Home_Automation_AUT@cluster0.9awkbpi.mongodb.net/?retryWrites=true&w=majority"
    client = MongoClient(uri)
    db = client['Home_Automation_DB']

    collection = db['devices']
    collection.update_one({"deviceName": my_device }, {"$set": {"status":f'{status}'}})
    print("Update Complete")

SENSOR_PIN = 17
RELAY_PIN = 18
REED_PIN = 22
LED_PIN = 27
IRO_PIN = 5
LED_PIN_IR = 24

GPIO.setmode(GPIO.BCM)

GPIO.setup(SENSOR_PIN, GPIO.IN)
GPIO.setup(RELAY_PIN, GPIO.OUT)
GPIO.setup(REED_PIN, GPIO.IN)
GPIO.setup(LED_PIN, GPIO.OUT)
GPIO.setup(IRO_PIN, GPIO.IN)
GPIO.setup(LED_PIN_IR, GPIO.OUT)

try:
    while True:
        sensor_value = GPIO.input(SENSOR_PIN)
        reed_value = GPIO.input(REED_PIN)
        ir_val = GPIO.input(IRO_PIN)

        if sensor_value == GPIO.HIGH:
            GPIO.output(RELAY_PIN, GPIO.LOW)
            update_document("office-watering")
        else:
            GPIO.output(RELAY_PIN, GPIO.HIGH)

        if reed_value == GPIO.HIGH:
            GPIO.output(LED_PIN, GPIO.HIGH)
        else:
            GPIO.output(LED_PIN, GPIO.LOW)

        if ir_val == GPIO.HIGH:
            GPIO.output(LED_PIN_IR, GPIO.HIGH)
        else:
            GPIO.output(LED_PIN_IR, GPIO.LOW)

        time.sleep(0.1)

except KeyboardInterrupt:
    GPIO.cleanup()